lib/content-services/src/lib/category/categories-management/categories-management.component.ts

```diff
@@ -0,0 +1,287 @@
+        ['duplicatedExistingCategory', 'ALREADY_EXISTS'],
+        ['duplicatedCategory', 'DUPLICATED_CATEGORY'],
+        ['emptyCategory', 'EMPTY_CATEGORY'],
+        ['required', 'REQUIRED']
+    ]);
+
+    private existingCategoryLoaded$ = new Subject<void>();
+    private cancelExistingCategoriesLoading$ = new Subject<void>();
+    private onDestroy$ = new Subject<void>();
+    private _categoryNameControl = new FormControl<string>(
+        '',
+        [
+            this.validateIfNotAlreadyAdded.bind(this),
+            this.validateEmptyCategory
+        ],
+        this.validateIfNotAlreadyCreated.bind(this)
+    );
+    private _existingCategories: Category[];
+    private _categoryNameErrorMessageKey = '';
+    private _existingCategoriesLoading = false;
+    private _typing = false;
+    private _existingCategoriesPanelVisible: boolean;
+    existingCategoriesListLimit = 15;
+    noCategoriesTitle = '';
+    removeCategoryTitle = '';
+    initialCategories: Category[] = [];
+
+    @Input()
+    categories: Category[] = [];
+
+    @Input()
+    categoryNameControlVisible = false;
+
+    @Input()
+    classifiableChanged: Observable<void>;
+
+    @Input()
+    disableRemoval = false;
+
+    @Input()
+    managementMode: CategoriesManagementMode;
+
+    @Input()
+    parentId: string;
+
+    @Output()
+    categoriesChange = new EventEmitter<Category[]>();
+
+    @Output()
+    categoryNameControlVisibleChange = new EventEmitter<boolean>();
+
+    constructor(private categoryService: CategoryService) {}
+
+    ngOnInit() {
+        this.categoryNameControl.valueChanges
+            .pipe(
+                map((name: string) => name.trim()),
+                tap((name: string) => {
+                    this._typing = true;
+                    if (name) {
+                        this._existingCategoriesLoading = true;
+                        this._existingCategoriesPanelVisible = true;
+                    } else {
+                        this._existingCategoriesPanelVisible = false;
+                    }
+                    this.cancelExistingCategoriesLoading$.next();
+                }),
+                debounce((name: string) => (name ? timer(300) : EMPTY)),
+                takeUntil(this.onDestroy$)
+            )
+            .subscribe((name: string) => this.onNameControlValueChange(name));
+
+        this.categoryNameControl.statusChanges
+            .pipe(takeUntil(this.onDestroy$))
+            .subscribe(() => this.setCategoryNameControlErrorMessageKey());
+
+        this.setCategoryNameControlErrorMessageKey();
+        this.noCategoriesTitle = this.isCRUDMode ?
+            'CATEGORIES_MANAGEMENT.NO_CATEGORIES_CREATED':
+            'CATEGORIES_MANAGEMENT.NO_CATEGORIES_ASSIGNED';
+        this.removeCategoryTitle = this.isCRUDMode ?
+            'CATEGORIES_MANAGEMENT.DELETE_CATEGORY':
+            'CATEGORIES_MANAGEMENT.UNASSIGN_CATEGORY';
```

Comment:
Small thing but I'm wondering if we need unassign label because in theory if you have categories which you want to assign then they are not yet assigned so technically you don't "unassign" them but you "remove" them from top list. But that's small thing, if you prefer that then that's fine to me :)

Reply:
Yes but it also depends on the case, since you can also start from having some categories assigned and then technically you're unassigning them so I think it would be more consistent to still call it unassign, we can discuss this with Agnieszka if you like

Reply:
As I said - it's small thing and I'm not sure too what is better, just wanted to highlight it :) We can keep that for now